pkgs/applications/science/logic/coq/default.nix

```diff
@@ -100,25 +112,14 @@ self = stdenv.mkDerivation {
     '';
   };
 
-  src = fetchFromGitHub {
-    owner = "coq";
-    repo = "coq";
-    rev = "V${version}";
-    inherit sha256;
-  };
-
-  nativeBuildInputs = [ pkgconfig ]
-  ++ stdenv.lib.optional (!versionAtLeast "8.6") gnumake42
-  ;
-  buildInputs = [ ncurses ocamlPackages.ocaml ocamlPackages.findlib ]
-  ++ stdenv.lib.optional (!versionAtLeast "8.10") ocamlPackages.camlp5
-  ++ stdenv.lib.optional (!versionAtLeast "8.12") ocamlPackages.num
-  ++ stdenv.lib.optionals buildIde
-    (if versionAtLeast "8.10"
-     then [ ocamlPackages.lablgtk3-sourceview3 glib gnome3.defaultIconTheme wrapGAppsHook ]
-     else [ ocamlPackages.lablgtk ]);
-
-  propagatedBuildInputs = stdenv.lib.optional (versionAtLeast "8.12") ocamlPackages.num;
+  nativeBuildInputs = [ pkgconfig ] ++ optional (!versionAtLeast "8.6") gnumake42;
+  buildInputs = [ ncurses ] ++ ocamlBuildInputs
+    ++ optional (!versionAtLeast "8.12") ocamlPackages.num
+    ++ optionals buildIde
+      (if versionAtLeast "8.10"
+       then [ ocamlPackages.lablgtk3-sourceview3 glib gnome3.defaultIconTheme wrapGAppsHook ]
+       else [ ocamlPackages.lablgtk ]);
+  propagatedBuildInputs = optional (versionAtLeast "8.12") ocamlPackages.num;
```

Comment:
IIUC, the `num` (or `zarith` from 8.13 onwards) library should be in `ocamlBuildInputs` and not propagated. The `mkCoqPackage` helper will take care of adding it to the `buildInputs` whenever needed (when the `ml-plugin` arguments is set).

This used to be propagated as a convenience in order to avoid regressions (see #94230).

Reply:
The doc is comming tomorrow, can we discuss this?

Reply:
Yes, because (from the linked PR discussion):

>if `ocamlfind` is told not to link a specific library, it tries to resolve it and fails if the library is not there

And thus, this propagated dependency is needed since https://github.com/coq/coq/pull/12604.

Reply:
But is it true that every coq package which relies on ocaml needs the `num` package? If it is true, then the debates ends there and I will simply add it :smile: However if it is not true, isn't it problematic to include more than the bare minimum? Shouldn't we let each package decide what they want to add?

Reply:
It's true because of this `ocamlfind` bug AFAIU.

Reply:
(only of plugins that build with `coq_makefile` though)

Reply:
And it's `num` up in 8.12 and `zarith` in 8.13+ (and no need before).

Reply:
Ok, I'll fix the thing tomorrow. Please do not hesitate to make other remarks so that I factor this in before I turn this PR from "Draft" to "Open"